apps/web/src/components/Messaging.tsx

```diff
@@ -0,0 +1,204 @@
+  if (!componentId) {
+    return '';
+  }
+
+  if (!componentId.id) {
+    return componentId.name;
+  }
+
+  return `${componentId.name.split('.').slice(1).join('.')}#${componentId.id}`;
+};
+
+const buildMessageSummary = (params: BWEMessage): ComponentMessage | null => {
+  const { toComponent, fromComponent, message } = params;
+  const isFromComponent = fromComponent !== undefined;
+  switch (message.type) {
+    case 'component.render': {
+      const { type, props } = message.node;
+      const formattedChildren = message.childComponents?.length
+        ? `with children ${message.childComponents
+            .map(({ componentId }) =>
+              formatComponentId(parseComponentId(componentId))
+            )
+            .join(', ')}`
+        : '';
+
+      return {
+        message,
+        isFromComponent,
+        name: 'render',
+        componentId: parseComponentId(message.componentId)!,
+        summary: `rendered <${type} ${formatProps(props, true).slice(
+          0,
+          64
+        )}...> ${formattedChildren}`,
+      };
+    }
+    case 'component.callbackInvocation': {
+      const targetComponent = formatComponentId(
+        parseComponentId(message.targetId || '')
+      );
+      const { requestId, method, args } = message;
+      return {
+        message,
+        isFromComponent,
+        name: 'invoke',
+        componentId: parseComponentId(message.originator)!,
+        summary: `[${requestId.split('-')[0]}] called ${targetComponent}.${
+          method.split('::')[0]
+        }(${args})${!isFromComponent ? ' for' : ''}`,
+      };
+    }
+    case 'component.callbackResponse': {
+      const { requestId, result } = message;
+      return {
+        message,
+        isFromComponent,
+        name: 'return',
+        componentId: parseComponentId(
+          isFromComponent ? message.containerId : message.targetId
+        )!,
+        summary: `[${requestId.split('-')[0]}] returned ${result} ${
+          !isFromComponent ? 'to' : ''
+        }`,
+      };
+    }
+    case 'component.update': {
+      return {
+        message,
+        isFromComponent,
+        name: 'update',
+        componentId: parseComponentId(toComponent!)!,
+        summary: `updated props ${JSON.stringify(message.props || {})} on`,
+      };
+    }
+    case 'component.domCallback': {
+      return {
+        message,
+        isFromComponent,
+        name: 'DOM',
+        componentId: parseComponentId(toComponent!)!,
+        summary: `invoked message DOM handler [${
+          message.method.split('::')[0]
+        }()] on`,
+      };
+    }
+    default:
+      return null;
+  }
+};
+
+export function Messaging({
+  containerMessages,
+}: {
+  containerMessages: BWEMessage[];
+}) {
+  const [selectedMessage, setSelectedMessage] = useState<object | null>(null);
+
+  const reversedMessages = [...containerMessages];
+  reversedMessages.reverse();
+
+  console.log({ selectedMessage });
```

Comment:
intentionally left in?

Reply:
Nope, just a very old branch I didn't vet very thoroughly 😅